frontend/src/app/patients/Components/PersonForm.tsx

```diff
@@ -116,6 +116,15 @@ const PersonForm = (props: Props) => {
 
   const schema = schemata[view];
 
+  // Language settings may persist into a non-i18nized view, so explicitly revert back to the
+  // default language in such cases
+  useEffect(() => {
+    if (schema !== selfRegistrationSchema) {
+      i18n.changeLanguage("en");
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
```

Comment:
That _is_ what I would think to do (and what eslint is yelling at me to do) but it introduces a problem on the person form:

> Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

Which, as you might expect, is accompanied by the freezing/crashing of the page.

My understanding was that the empty dependency array would cause the hook to run only once, thereby acting as a sort of `componentDidMount`. Does that seem OK, or is the best practice for this pattern escaping me?

Reply:
In that case I'd _still_ include the required dependencies but just do a check before changing the language. My guess is the language change is causing new schema objects to be created every time, which is causing infinite re-renders:

I _think_ something like this should work:

```javascript
  useEffect(() => {
    if (i18n.language !== "en" && schema !== selfRegistrationSchema) {
      i18n.changeLanguage("en");
    }
  }, [schema, selfRegisitrationSchema]);
```

Reply:
Yup, that does the trick. Updated in f50c703.